README.md

```diff
@@ -19,66 +19,76 @@ pip3 install gmqtt
 Here is a very simple example that subscribes to the broker TOPIC topic and prints out the resulting messages:
 
 ```python
+#!/usr/bin/python3
+
 import asyncio
 import os
 import signal
 import time
 
-from gmqtt import Client as MQTTClient
-
-# gmqtt also compatibility with uvloop  
-import uvloop
-asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
-
-
-STOP = asyncio.Event()
+import gmqtt
 
 
 def on_connect(client, flags, rc, properties):
     print('Connected')
     client.subscribe('TEST/#', qos=0)
 
-
 def on_message(client, topic, payload, qos, properties):
     print('RECV MSG:', payload)
 
-
 def on_disconnect(client, packet, exc=None):
     print('Disconnected')
 
 def on_subscribe(client, mid, qos, properties):
     print('SUBSCRIBED')
 
-def ask_exit(*args):
-    STOP.set()
-
-async def main(broker_host, token):
-    client = MQTTClient("client-id")
-
+def create_client(token):
+    client = gmqtt.Client('client-id')
     client.on_connect = on_connect
     client.on_message = on_message
     client.on_disconnect = on_disconnect
     client.on_subscribe = on_subscribe
-
     client.set_auth_credentials(token, None)
-    await client.connect(broker_host)
+    return client
 
+async def publish(client):
     client.publish('TEST/TIME', str(time.time()), qos=1)
 
-    await STOP.wait()
-    await client.disconnect()
-
+def shutdown(loop, client):
+    loop.run_until_complete(client.disconnect())
+    # The disconnection creates new tasks, so perform another run of
+    # the loop to flush out those tasks.
+    async def flush():
+        pass
+    loop.run_until_complete(flush())
 
 if __name__ == '__main__':
-    loop = asyncio.get_event_loop()
 
     host = 'mqtt.flespi.io'
     token = os.environ.get('FLESPI_TOKEN')
 
-    loop.add_signal_handler(signal.SIGINT, ask_exit)
-    loop.add_signal_handler(signal.SIGTERM, ask_exit)
+    client = create_client(token)
+
+    loop = asyncio.get_event_loop()
+    loop.add_signal_handler(signal.SIGINT, loop.stop)
+    loop.add_signal_handler(signal.SIGTERM, loop.stop)
+
+    t = loop.create_task(client.connect(host))
+    # Inject a message, once the connection is established.
+    def publish(fut):
+        client.publish('TEST/TIME', str(time.time()), qos=1)
+    t.add_done_callback(publish)
```

Comment:
More explicit way to do so is to place publish in on_connect callback :slightly_smiling_face:

Reply:
Thanks for the idea. Moved!

Reply:
An unexpected change: upon a **reconnect**, another test message is published. The "done callback" approach only publishes a single message. ... That said, I prefer the on_connect approach for **sample code**, as it is simpler to understand for the reader.